### DISCUSSION TITLE
Breakpoints on method return statements

### DISCUSSION THREAD

**Discussion by Connell1** [dotnet/roslyn#49466] [Ideas] [answered]
While debugging in C#, I often find myself taking a method of the form:

`string DoSomething(object val) { return SomeOtherMethod(val); }`

and breaking it into multiple lines like:

`string DoSomething(object val) { var result = SomeOtherMethod(val); return result; }`

just so that I can set a breakpoint on the `return` and see the value. Would it be possible to make it so that on Debug builds, a breakpoint on a return statement would show the value that will be returned?

**Comment by canton7** [comment, marked as answer]
If you step onto the closing brace, you can see the return value in the Locals window, and you can also put `$ReturnValue` into the Watch window:

![image](https://user-images.githubusercontent.com/568104/99546798-b3a64a00-29ae-11eb-9239-bc15f4251a41.png)

However, this doesn't seem to happen if you put a breakpoint explicitly on the closing brace: you have to step onto the closing brace.

You can also see the return value in the Locals window after the method has returned:

![image](https://user-images.githubusercontent.com/568104/99547061-03851100-29af-11eb-8d62-a7e3a9d43daf.png)

**Reply by Connell1** [reply]
Thanks
